### PR TITLE
Config `hatch` for release workflow in GHA

### LINKED ISSUE #4: Add github actions release workflow
Need to add a workflow similar to what we have in other repos to handle bumping version and releasing
example: https://github.com/dbt-labs/dbt-bigquery/actions/workflows/release.yml

```[tasklist]
- [x] Release to PyPI
- [x] Release to PyPI Test
- [ ] Release to GitHub
- [ ] Release Docker
- [ ] Add version bumping
```

### PR DESCRIPTION
**Pull request by mikealfare**
resolves #4

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-adapter/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] This PR has no interface changes (e.g. macros, cli, logs, json artifacts, config files, adapter interface, etc) or this PR has already received feedback and approval from Product or DX

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+
+      - name: Setup environment
+        uses: ./.github/actions/setup-environment
+        with:
+          python-version: "3.8"
```
Comment (emmyoop) [inline .github/workflows/release.yml:39]: Can we use anything later than 3.8 please.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+      - name: Build artifacts
+        run: hatch build
+
+      - name: Check artifacts
+        run: hatch run check-all
```
Comment (emmyoop) [inline .github/workflows/release.yml:45]: Nice!

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+      - name: Publish artifacts
+        run: hatch publish -r ${{ inputs.pypi }}
```
Comment (emmyoop) [inline .github/workflows/release.yml:48]: This won't work for auth.  You need to use the action.

[pypa/gh-action-pypi-publish@v1](https://github.com/pypa/gh-action-pypi-publish?tab=readme-ov-file#pypi-publish-github-action)

**Review by emmyoop — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (mikealfare) [reply .github/workflows/release.yml]: I updated to py311

File: .github/workflows/release.yml (hunk 3 above)
Comment (mikealfare) [reply .github/workflows/release.yml]: I updated this and put it in a shared action for readability. We'll need to add one for GH and potentially others later, so it makes the main workflow shorter.

**Review by colin-rogers-dbt — APPROVED** [review]

**Review by colin-rogers-dbt — APPROVED** [review]

**Pull request merged**